lib/OpenQA/Script/CloneJob.pm

```diff
@@ -221,54 +226,60 @@ sub handle_tx ($tx, $jobid, $job, $base_url, $options, $clone_map, $depth, $chil
     }
 }
 
-sub clone_job ($jobid, $options, $clone_map = {}, $depth = 0, $parent_jobid = 0) {
-    return $clone_map->{$jobid} if defined $clone_map->{$jobid};
-
-    my ($ua, $local, $local_url, $remote, $remote_url) = create_url_handler($options);
-    my $job = clone_job_get_job($jobid, $remote, $remote_url, $options);
-    my @job_types = ('parents');
-    push @job_types, 'children' if $options->{'clone-children'} && $depth == 0;
-    my $child_list;
-    for my $job_type (@job_types) {
-        if ($job->{$job_type}) {
-            my ($chained, $directly_chained, $parallel) = get_deps($job, $options, $job_type);
-
-            print "Cloning dependencies of $job->{name}\n" if (@$chained || @$directly_chained || @$parallel);
-            push @$child_list, @$parallel if (@$parallel && $job_type eq 'children');
-            for my $dependencies ($chained, $directly_chained, $parallel) {
-                # don't clone parallel jobs yet if children job type
-                next if $dependencies == $parallel && $job_type eq 'children';
-                clone_job($_, $options, $clone_map, $depth + 1) for @$dependencies;
-                # abort here if the job has already been cloned as part of the preceding recursive clone_job call
-                return $clone_map->{$jobid} if defined $clone_map->{$jobid};
-            }
+sub clone_jobs ($jobid, $options) {
+    my $url_handler = create_url_handler($options);
+    clone_job($jobid, $url_handler, $options, my $post_params = {}, my $jobs = {});
+    my $tx = post_jobs($post_params, $url_handler, $options);
+    handle_tx($tx, $url_handler, $options, $jobs);
+}
+
+sub clone_job ($jobid, $url_handler, $options, $post_params = {}, $jobs = {}, $depth = 1) {
```

Comment:
`$depth` is hardcoding here.

Reply:
that's not hardcoding but just specifying a default value.

Reply:
my bad. i rush to add the comment. ignore